src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -732,13 +733,14 @@ public static void removeUnusedSstables(CFMetaData metadata, Map<Integer, UUID>
                 assert compactionTaskID != null;
                 logger.info("Going to delete unfinished compaction product {}", desc);
                 SSTable.delete(desc, allNonTempSstableFiles.get(desc));
-                SystemKeyspace.finishCompaction(compactionTaskID);
+                cleanedCompactions.add(compactionTaskID);
             }
             else
             {
                 completedAncestors.addAll(ancestors);
             }
         }
+        cleanedCompactions.forEach(SystemKeyspace::finishCompaction);
 
         // remove old sstables from compactions that did complete
```

Comment:
necessary so if we crash mid-startup we won't nuke the compaction log and corrupt anyway

Reply:
And we don't make contrary cleanup decisions on "single compaction create multiple sstables" resulting sstables that still have their relevant compaction id in the `compaction_in_progress` table